File: plugins/architecture-tests/evals/fixtures/tsarch/src/domain/order.ts
// Domain layer: the pure core. It INTENTIONALLY imports Infrastructure, which is the
// architecture violation the locked ts-arch contract must detect (Domain -> Infrastructure).
import { Database } from "../infrastructure/database";

export class Order {
  private readonly db = new Database();

  place(id: string): void {
    this.db.save(id);
  }
}

File: plugins/architecture-tests/evals/fixtures/netarchtest/src/Sample.Fixture/Domain/Order.cs
using Sample.Infrastructure;

namespace Sample.Domain
{
    // INTENTIONAL ARCHITECTURE VIOLATION: a Domain type must not depend on the Infrastructure layer,
    // yet Order references Sample.Infrastructure.Database. The NetArchTest rule detects this edge and fails.
    public class Order
    {
        private readonly Database _db = new Database();

        public void Persist()
        {
            _db.Save("order");
        }
    }
}
